src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -700,7 +715,9 @@ private void validateShare(Path sharePath) throws IOException {
     // Verify that the adaptor has permission to read the Acl and share Acl.
     try {
       readShareAcls(sharePath);
-      delegate.getAclViews(sharePath);
+      if (!delegate.isDfsNamespace(sharePath) || allowFilesInDfsNamespaces) {
```

Comment:
If we do have problems treating DFS Namespaces as regular directories, I can imagine reading the Namespace's filesystem ACLs would be a likely one.  This change is a bit defensive.  It verifies we can read the ACL if treating Namespaces as directories (so we get the error at init time), but maintains the previous behavior of not attempting to do so when not treating Namespaces as directories.